### PR TITLE
HPCC-30218 Use default spray target if not specified in cloud

### PR DESCRIPTION
**Pull request by wangkx**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: ecl/eclagent/eclagent.cpp [hunk 1]
```diff
@@ -3129,7 +3129,7 @@ char * EclAgent::getGroupName()
 {
 #ifdef _CONTAINERIZED
     // in a containerized setup, the group is moving..
-    return strdup("unknown");
+    return strdup("");
```
Comment (jakesmith) [inline ecl/eclagent/eclagent.cpp:3132]: let's revise the comment here now the interpretation has changed.
Replace the comment above with:

// In containerized there is no associated cluster group.
// Return an empty result instead, which if used for destinationGroup
// via Thorlib.group() as part of a fileservices call, will be interpreted
// as targeting the default data plane

File: esp/services/ws_fs/ws_fsService.cpp [hunk 2]
```diff
@@ -2083,15 +2083,28 @@ static void checkValidDfuQueue(const char * dfuQueue)
...
 
+static const char* getSprayDestGroup(const char* groupReq, StringBuffer& groupOut)
+{
+    if (!isEmptyString(groupReq))
+        groupOut.append(groupReq);
+    else if (isContainerized())
+    {
+        Owned<IPropertyTreeIterator> dataPlanes = getGlobalConfigSP()->getElements("storage/planes[@category='data']");
+        if (dataPlanes->first())
+            groupOut.append(dataPlanes->query().queryProp("@name"));
```
Comment (jakesmith) [inline esp/services/ws_fs/ws_fsService.cpp:2094]: the above should use getDefaultStoragePlane

**Review by jakesmith — COMMENTED** [review]
@wangkx - looks functionally right, but see comments

Did you manage to verify/test this in both bare-metal and cloud with a file service call from ECL with destinationGroup=STD.System.Thorlib.Group() ?

**Comment by wangkx** [comment]
@jakesmith I made the changes based on your comments. It tested in both bare-metal and cloud with a file service call from ECL with destinationGroup=STD.System.Thorlib.Group().

**Review by jakesmith — APPROVED** [review]
@wangkx - looks good

**Pull request merged**